### PR TITLE
Add sparse workspace features support to the indirect correction UI

### LINKED ISSUE #34716: Add fields to support sparse workspace features in Indirect, Corrections UI
The "Calculate Monte Carlo Absorption" tab in the Indirect, Corrections UI calls the PaalmanPingsMonteCarloAbsorption algorithm to perform an absorption correction. This algorithm supports a sparse workspace feature which calculates the correction on a subset of the detectors on an instrument and then uses interpolation to return a result for the remaining detectors. This speeds up the calculation on instruments with a lot of detectors.

This change is to add the sparse workspace fields to the UI so the users can take advantage of this feature. The following screenshot of the algorithm dialog for PaalmanPingsMonteCarloAbsorption shows the required fields:

image

Requested by Spencer Howells

### PR DESCRIPTION
**Pull request by MohamedAlmaki**
**Description of work.**
This PR adds sparse workspace features support to the "Calculate Monte Carlo Absorption" tab in the Indirect Corrections UI. Three fields have been added to the UI spare instrument checkbox, the number of detector rows and the number of detector columns. Note, that these fields are already present in the PaalmanPingsMonteCarloAbsorption algorithm UI, so I just added them to the correction UI.

**Report to:** Spencer Howells

**To test:**

From the top bar open Interfaces->Indirect->Corrections. Then open Calculate Monte Carlo Absorption tab. You will find the new fields added in the Monte Carlo section. After that, select a sample workspace and input some values in the new fields then click run. Finally, check the history of the output to ensure the fields are recorded in the history of the algorithms. 

Fixes #34716. 

*This does not require release notes* because **it is a simple UI modification**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/AbsorptionCorrections.ui [hunk 1]
```diff
@@ -158,6 +158,62 @@
...
+           <widget class="QLabel" name="lbNumberDetectorColumns">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
+            <property name="text">
+             <string>Number Of Detector Columns:</string>
+            </property>
+           </widget>
+          </item>
+          <item row="4" column="3">
```
Comment (DannyHindson) [inline qt/scientific_interfaces/Indirect/AbsorptionCorrections.ui:204]: Minor point - I think the columns spinbox should be aligned with the "Max Scatter Point Attempts" above it. You can achieve this by changing the column to "4" here:

image

**Review by DannyHindson — CHANGES_REQUESTED** [review]
I've tested this and the new fields are being passed through to the algorithm correctly. 
Just one minor point about the alignment of one of the new spin boxes. Plus I think this probably does need a release note even though it's just a UI change so please can you add one. It can go in `docs\source\release\v6.6.0\Indirect\New_features`

**Review by DannyHindson — CHANGES_REQUESTED** [review]
One more thing. There's a documentation page here: `docs\source\interfaces\indirect\Indirect Corrections.rst` that describes all the fields on the corrections UI. Please can you add a few lines at ~L290 in here to describe the new fields? I think the screenshot on the same page is automatically generated so no work required there.

File: qt/scientific_interfaces/Indirect/AbsorptionCorrections.ui (hunk 1 above)
Comment (MohamedAlmaki) [reply qt/scientific_interfaces/Indirect/AbsorptionCorrections.ui:204]: Ok, will fix this.

**Comment by MohamedAlmaki** [comment]
> One more thing. There's a documentation page here: `docs\source\interfaces\indirect\Indirect Corrections.rst` that describes all the fields on the corrections UI. Please can you add a few lines at ~L290 in here to describe the new fields? I think the screenshot on the same page is automatically generated so no work required there.

Ok, will add the release note and edit the documentation. Thanks for your review.

**Review by DannyHindson — CHANGES_REQUESTED** [review]
Thanks for the release note and aligning the new edit box. Just one small comment about the new lines in the indirect UI documentation

**Comment by DannyHindson** [comment]
Hi Mohamed, just want to check we're not waiting on each other on this PR. I saw you did another push on this yesterday but seems like you didn't include any change to the Indirect Corrections.rst file? The change I was after was to simply modify "DetectorRows" and "DetectorColumns" to "Detector Rows" and "Detector Columns" respectively. Thanks

**Comment by MohamedAlmaki** [comment]
> Hi Mohamed, just want to check we're not waiting on each other on this PR. I saw you did another push on this yesterday but seems like you didn't include any change to the Indirect Corrections.rst file? The change I was after was to simply modify "DetectorRows" and "DetectorColumns" to "Detector Rows" and "Detector Columns" respectively. Thanks

Oh, sorry. I rebased the branch to the main and didn't see your comment. Will update the doc right now.

**Review by DannyHindson — APPROVED** [review]

**Pull request merged**